### PR TITLE
[Feature][MRV2] Support DSA prefill context parallelism

### PR DESCRIPTION
**Pull request by li1how**
### What this PR does / why we need it?

This PR adds DeepSeek-V4 DSA prefill context parallelism support to
ModelRunner V2 while preserving the existing standalone DSA and legacy
DSACP paths.

- Deduplicate the Q projection, SWA cache update, and multistream QLI
  completion shared by DSA prefill and decode.
- Add a cache-prepared execution mode that separates global SWA,
  compressor, and indexer cache updates from rank-local attention.
- Add a PCP-specific DSA backend and metadata builder that derive a
  canonical global token layout, gather fixed-shape hidden states, and
  replay the main and indexer compressors in the same order on every PCP
  rank.
- Run attention on rank-local tokens after cache preparation and restore
  rank-local outputs, while rejecting unsupported PCP combinations.

### Does this PR introduce _any_ user-facing change?

Yes. DeepSeek-V4 DSA models running with ModelRunner V2 can use prefill
context parallelism through the existing PCP configuration. This PR does
not add new CLI arguments or configuration fields.

### How was this patch tested?

DeepSeek-V4 Flash W4A8, 128K prefill (131072 input tokens + 1 output token),
16 NPUs, DP2:

| Global batch | TP8 baseline | DSACP | PCP8 | PCP8 vs DSACP |
| ---: | ---: | ---: | ---: | ---: |
| 1 | 7.88k tok/s | 12.49k tok/s | 15.12k tok/s | +21.05% |
| 2 | 14.60k tok/s | 22.50k tok/s | 24.37k tok/s | +8.32% |
| 4 | 15.28k tok/s | 23.86k tok/s | 26.01k tok/s | +8.98% |
| 8 | 15.77k tok/s | 25.22k tok/s | 27.35k tok/s | +8.44% |

GPQA Diamond first 100 samples, 8 NPUs, DP1: PCP8 accuracy is **62%**,
compared with **60%** for the TP8 baseline (**+2 pp**); both completed
100/100 requests successfully.

- vLLM version: v0.27.1
- vLLM main: https://github.com/vllm-project/vllm/commit/cdc4824a21eaa986d4d1fee90a7e6465c9f706e6

### CODE REVIEW

File: vllm_ascend/attention/dsa_v1.py [hunk 1]
```diff
@@ -914,6 +987,293 @@ def build_for_graph_capture(
...
+            token_start = int(query_start_loc_cpu[row])
+            num_tokens = int(query_lens_cpu[row])
+            start_pos = int(seq_lens_cpu[row]) - num_tokens
+            local_segment_descriptors_cpu[row] = torch.tensor(
+                (token_start, num_tokens, start_pos, 1),
+                dtype=torch.int32,
+            )
+        segment_descriptors_by_rank = (
+            get_pcp_group()
+            .all_gather(
```
Comment (pisceskkk) [inline vllm_ascend/attention/dsa_v1.py]: Do we need all_gather in MetadataBuilder?

File: vllm_ascend/attention/context_parallel/dsa_cp.py [hunk 2]
```diff
@@ -1828,3 +1839,381 @@ def _indexer_select_topk(
...
         return topk_idxs
+
+
+# =============================================================================
+# MRV2 DSA-PCP implementation
+# =============================================================================
+
+
+@dataclass(kw_only=True)
+class AscendDSAPCPMetadata(dsa_v1.AscendDSAMetadata):
```
Comment (weiguihua2) [inline vllm_ascend/attention/context_parallel/dsa_cp.py:1930]: Supplement PCP information

**Review by weiguihua2 — DISMISSED** [review]

File: vllm_ascend/attention/context_parallel/dsa_cp.py (hunk 2 above)
Comment (li1how) [reply vllm_ascend/attention/context_parallel/dsa_cp.py:1930]: Added PCP documentation in [#14571](https://github.com/vllm-project/vllm-ascend/pull/14571). It covers MRV2 usage and constraints, DualChunkSwap partitioning, and the GQA/DSA cache-preparation paths. The documentation PR depends on this PR for the DSA PCP section.

**Review by pisceskkk — APPROVED** [review]

**Comment by li1how** [comment]
/rerun
[Bot]: rerun completed.

**Failed:**
- E2E: some jobs could not be re-run (still waiting on dependencies)

File: vllm_ascend/attention/dsa_v1.py [hunk 3]
```diff
@@ -90,10 +91,18 @@ def get_name() -> str:
     def get_builder_cls():
         from vllm_ascend.utils import enable_dsa_cp
 
-        if enable_dsa_cp():
+        use_dsa_cp = enable_dsa_cp()
+        use_pcp = enable_pcp()
+        if use_dsa_cp and use_pcp:
```
Comment (zhenwenqi2024) [inline vllm_ascend/attention/dsa_v1.py]: dsa_cp 被下架了，不会有这种配置了吧

**Comment by li1how** [comment]
/rerun
[Bot]: rerun completed.

**Rerun (failed jobs only):**
- E2E

**Review by ZT-AIA — APPROVED** [review]
